pkg/runtime/optimize/optimizer.go

```diff
@@ -169,6 +171,24 @@ func (o optimizer) optimizeDropIndex(ctx context.Context, stmt *rast.DropIndexSt
 	return shardPlan, nil
 }
 
+func (o optimizer) optimizeCreateIndex(ctx context.Context, stmt *rast.CreateIndexStatement, args []interface{}) (proto.Plan, error) {
+	ru := rcontext.Rule(ctx)
+	//table shard
+
+	shard, err := o.computeShards(ru, stmt.Table, nil, args)
+	if err != nil {
+		return nil, err
+	}
+	if len(shard) == 0 {
+		return plan.Transparent(stmt, args), nil
+	}
```

Comment:
I think this is not the correct way to get shards.
1. check whether the `stmt.Table` is a logical table, bypass if not.
2. enumerate all shards from the logical table topology, see the example code:

```go
		// init shards
		shards := rule.DatabaseTables{}
		// compute all tables
		topology := vt.Topology()
		topology.Each(func(dbIdx, tbIdx int) bool {
			if d, t, ok := topology.Render(dbIdx, tbIdx); ok {
				shards[d] = append(shards[d], t)
			}
			return true
		})
```

Reply:
Hello, may I ask you a question, how to check  whether the stmt.Table is a logical table ?

Reply:
See the snippet below:
```go
var ru *rule.Rule // prepare your rule
_, ok := ru.VTable("your_table_name")
// target is logical table if ok is true.
```